### DISCUSSION TITLE
Snowflake Connection issue

### DISCUSSION THREAD

**Discussion by SubratSrivastwa** [hasura/graphql-engine#9352] [Question] [answered]
Hi team ,

I am getting below error : -
Inconsistent object: UncaughtError: Unable to get java.sql.Driver from DriverManager 

When I am trying to connect to my snowflake sandbox account 

Thanks ,
Subrat

**Comment by meetzaveri** [comment, marked as answer]
Hey @SubratSrivastwa 

This is a known issue from our end, and we are currently working on a fix. Also, we are in the process of updating docs for better developer experience regarding snowflake integration. For now, you can try the below mentioned workaround and check if it solves your issue.

### Workaround
For the database URL, it currently expects a JDBC URL. Here is an example one with some common options set:
 
```
jdbc:snowflake://{ACCOUNT_IDENTIFIER}.snowflakecomputing.com?user={USER}&password={PASSWORD}&warehouse={WAREHOUSE}&db={DB}&role={ROLE}&schema={DB_SCHEMA}
```
 
Information about `ACCOUNT_IDENTIFIER` is in the Snowflake docs [here](https://docs.snowflake.com/en/user-guide/jdbc-configure.html#jdbc-driver-connection-string). Information about the connection parameters USER, PASSWORD, WAREHOUSE, DB, ROLE, SCHEMA, and any other options is in the Snowflake docs [here](https://docs.snowflake.com/en/user-guide/jdbc-parameters.html#jdbc-driver-connection-parameter-reference).

Please let me know if you run into any other problems or if there is any other question I can try and help answer. I have already escalated to the responsible team to have a look at this and provide you an update.

**Comment by dameleney** [comment]
@SubratSrivastwa This issue should now be resolved are you still getting an error message when trying to connect Hasura to your Snowflake Sandbox?

**Reply by csteenbock** [reply]
I am getting this error when using oAuth 2.0 client credentials flow via Entra ID. I have verified that my client ID, secret, scope, and token URL are all correct, as I can use them from a python app just fine.
